src/main/java/org/jboss/threads/StripedEnhancedQueueExecutor.java

```diff
@@ -0,0 +1,736 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2020 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.threads;
+
+import static org.jboss.threads.EnhancedQueueExecutor.DEFAULT_HANDLER;
+import static org.jboss.threads.EnhancedQueueExecutor.REGISTER_MBEAN;
+import static org.jboss.threads.EnhancedQueueExecutor.TS_THREAD_CNT_MASK;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.jboss.threads.management.ManageableThreadPoolExecutorService;
+import org.jboss.threads.management.StandardThreadPoolMXBean;
+import org.wildfly.common.Assert;
+
+/**
+ * A striped version of {@link EnhancedQueueExecutor}.
+ */
+public final class StripedEnhancedQueueExecutor extends AbstractExecutorService implements ManageableThreadPoolExecutorService {
+    private final List<EnhancedQueueExecutor> executors;
```

Comment:
We may want to use either an array, or take advantage of the random-access property of the array list to avoid iterator allocation in accessors.

Reply:
Good idea.